### ISSUE TITLE
Open images regardless their extension

### ISSUE THREAD

**Issue by Atcold** [torch/image#129]
I was wondering if / how `image` could be tweaked for opening images regardless their extension. This would be extremely helpful when `JPEG`s have a `PNG` extension or the other way around.
I was thinking perhaps to check the `st_mode` field of the structure `struct stat` returned by `STAT(2)`, perhaps using `ffi`, but I don't know if it would be the best approach.
Please advice.

**Comment by Atcold** [comment]
Or, what about having the opening done entirely in C, so that it checks the format and then chooses how to open the file? It should not take much more time, shouldn't it?

**Comment by deltheil** [comment]
You can do that at the user level by checking the magic (like [this](https://github.com/torch/image/blob/0e43c1e/init.lua#L100-L111)), e.g.:

```Lua
require 'image'
require('pl.stringx').import()

local p = 'foo.jpg'
local f = io.open(p, 'r')
local h = f:read(4)
f:close()

local load

if h:startswith(string.char(0xff, 0xd8, 0xff)) then
  load = image.loadJPG
elseif h:startswith(string.char(0x89,0x50,0x4e,0x47)) then
  load = image.loadPNG
end

assert(load, 'format not supported')

local img = load(p)
```

**Comment by Atcold** [comment]
Aww :heart::heart::heart:
Shouldn't this be done by default? Or add a flag (`checkFileType` == `true`) to the current `image.load()`?

**Comment by deltheil** [comment]
Solved by #140.

**Comment by Atcold** [comment]
Thank you @deltheil!
